akka-docs-dev/rst/java/stream-customize.rst

```diff
@@ -9,6 +9,201 @@ is sometimes necessary to define new transformation stages either because some f
+It is possible to opt out from this behavior by overriding ``keepGoingAfterAllPortsClosed`` and returning true in
+the :class:`GraphStageLogic` implementation. In this case the stage can be explicitly closed by calling ``completeStage()``
+or ``failStage(exception)``. This feature has the risk of leaking streams and actors, therefore it should be used
+with care.
+
+Using timers
+------------
+
+**This section is a stub and will be extended in the next release**
+
+It is possible to use timers in :class:`GraphStages` by using :class:`TimerGraphStageLogic` as the base class for
+the returned logic. Timers can be scheduled by calling one of ``scheduleOnce(key,delay)``, ``schedulePeriodically(key,period)`` or
+``schedulePeriodicallyWithInitialDelay(key,delay,period)`` and passing an object as a key for that timer (can be any object, for example
+a :class:`String`). The ``onTimer(key)`` method needs to be overridden and it will be called once the timer of ``key``
+fires. It is possible to cancel a timer using ``cancelTimer(key)`` and check the status of a timer with
+``isTimerActive(key)``. Timers will be automatically cleaned up when the stage completes.
+
+Using asynchronous side-channels
+--------------------------------
+
+**This section is a stub and will be extended in the next release**
+
+In order to receive asynchronous events that are not arriving as stream elements (for example a completion of a future
+or a callback from a 3rd party API) one must acquire a :class:`AsyncCallback` by calling ``getAsyncCallback()`` from the
+stage logic. The method ``getAsyncCallback`` takes as a parameter a callback that will be called once the asynchronous
+event fires. It is important to **not call the callback directly**, instead, the external API must call the
+``invoke(event)`` method on the returned :class:`AsyncCallback`. The execution engine will take care of calling the
+provided callback in a thread-safe way. The callback can safely access the state of the :class:`GraphStageLogic`
+implementation.
+
+Integration with actors
+-----------------------
+
+**This section is a stub and will be extended in the next release**
+**This is an experimental feature***
+
+It is possible to acquire an ActorRef that can be addressed from the outside of the stage, similarly how
+:class:`AsyncCallback` allows injecting asynchronous events into a stage logic. This reference can be obtained
+by calling ``getStageActorRef(receive)`` passing in a function that takes a :class:`Pair` of the sender
+:class:`ActorRef` and the received message. This reference can be used to watch other actors by calling its ``watch(ref)``
+or ``unwatch(ref)`` methods. The reference can be also watched by external actors. The current limitations of this
+:class:`ActorRef` are:
+
+ - they are not location transparent, they cannot be accessed via remoting.
+ - they cannot be returned as materialized values.
```

Comment:
Maybe also:

they can not be obtained in the Logic's constructor. The earliest they can be initialized is during the Stage's preStart